src/main/java/gregtech/api/recipes/ingredients/GTRecipeItemInput.java

```diff
@@ -7,11 +7,31 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.NonNullList;
 
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 public class GTRecipeItemInput extends GTRecipeInput {
-    ItemStack[] inputStacks;
-    List<ItemToMetaList> itemList = new ObjectArrayList<>();
+
+    public static GTRecipeInput getOrCreate(ItemStack stack, int amount) {
+        return new GTRecipeItemInput(stack, amount);
+    }
+
+    public static GTRecipeInput getOrCreate(GTRecipeInput ri, int i) {
+        return new GTRecipeItemInput(ri.getInputStacks(), i);
+    }
+
+    public static GTRecipeInput getOrCreate(GTRecipeInput ri) {
+        return new GTRecipeItemInput(ri.getInputStacks());
+    }
+
+    public static GTRecipeInput getOrCreate(ItemStack ri) {
+        return new GTRecipeItemInput(ri);
+    }
```

Comment:
These methods don't do any form of "getting." Would it be better to simply call them `create()`? This could also apply to all of the `getOrCreate()` GTRecipeInput methods.

Reply:
I agree that the method name does not fit, but we can't delete them because they're already being exposed as API.